$MirOS: ports/lang/llvm-gcc/patches/patch-libcpp_macro_c,v 1.1 2008/11/10 20:30:56 tg Exp $

	Security fix for 64-bit time_t

--- libcpp/macro.c.orig	Wed Sep  2 20:47:05 2009
+++ libcpp/macro.c	Sat Dec  5 14:23:52 2009
@@ -239,11 +239,13 @@ _cpp_builtin_macro_text (cpp_reader *pfi
 
 	  if (tb)
 	    {
+	      size_t pfile_date_sz;
 	      pfile->date = _cpp_unaligned_alloc (pfile,
+						  pfile_date_sz =
 						  sizeof ("\"Oct 11 1347\""));
-	      sprintf ((char *) pfile->date, "\"%s %2d %4d\"",
+	      snprintf ((char *) pfile->date, pfile_date_sz, "\"%s %2d %4d\"",
 		       monthnames[tb->tm_mon], tb->tm_mday,
-		       tb->tm_year + 1900);
+		       (int)(tb->tm_year + 1900));
 
 	      pfile->time = _cpp_unaligned_alloc (pfile,
 						  sizeof ("\"12:34:56\""));
